[2025-08-22 Fiddler](https://thefiddler.substack.com/p/how-far-can-you-run-before-sundown)
====================

Fiddler
-------
I will run 6.5 miles.

In [1]:
S10 = 1/4
S15 = 1/4
S20 = (1 + S10)/4
S25 = (1 + S15)/4
S30 = (1 + S20 + 3)/4
S35 = (1 + S25 + 3 + 3+1/2)/4
S45 = (1 + S35 + 3 + S15 + 3+1/2 + S10 + 4+1/2)/4
S55 = (1 + S45 + 3 + S25 + 3+1/2 + S20 + 4+1/2 + S10)/4
S65 = (1 + S55 + 3 + S35 + 3+1/2 + S30 + 4+1/2 + S20)/4
S65, numerical_approx(S65)

(19933/4096, 4.86645507812500)

And my average score will be approximately 4.866 miles.

Extra credit
------------

In [2]:
S10m = (max(1,S10) + 3*max(0,S10))/4
S15m = (max(1,S15) + 3*max(0,S15))/4
S20m = (max(1+S10m,S20) + 3*max(0,S20))/4
S25m = (max(1+S15m,S25) + 3*max(0,S25))/4
S30m = (max(1+S20m,S30) + max(3,S30) + 2*max(0,S30))/4
S35m = (max(1+S25m,S35) + max(3,S35) + max(3+1/2,S35) + max(0,S35))/4
S45m = (max(1+S35m,S45) + max(3+S15m,S45) + max(3+1/2+S10m,S45) + max(4+1/2,S45))/4
S55m = (max(1+S45m,S55) + max(3+S25m,S55) + max(3+1/2+S20m,S55) + max(4+1/2+S10m,S55))/4
S65m = (max(1+S55m,S65) + max(3+S35m,S65) + max(3+1/2+S30m,S65) + max(4+1/2+S20m,S65))/4
S65m,numerical_approx(S65m)

(21921/4096, 5.35180664062500)

And my average score will be approximately 5.352 miles.

### When to use the mulligan

In [3]:
print("1",(1,numerical_approx(S10)))
print("1.5",(1,numerical_approx(S15)))
print("2",(numerical_approx(1+S10m),numerical_approx(S20)))
print("2.5",(numerical_approx(1+S15m),numerical_approx(S25)))
print("3",(numerical_approx(1+S20m),3,numerical_approx(S30)))
print("3.5",(numerical_approx(1+S25m),3,3.5,numerical_approx(S35)))
print("4.5",(numerical_approx(1+S35m),numerical_approx(3+S15m),
             numerical_approx(3.5+S10m),4.5,numerical_approx(S45)))
print("5.5",(numerical_approx(1+S45m),numerical_approx(3+S25m),
             numerical_approx(3.5+S20m),numerical_approx(4.5+S10m),numerical_approx(S55)))
print("6.5",(numerical_approx(1+S55m),numerical_approx(3+S35m),
             numerical_approx(3.5+S30m),numerical_approx(4.5+S20m),numerical_approx(S65)))

1 (1, 0.250000000000000)
1.5 (1, 0.250000000000000)
2 (1.43750000000000, 0.312500000000000)
2.5 (1.43750000000000, 0.312500000000000)
3 (1.59375000000000, 3, 1.07812500000000)
3.5 (1.59375000000000, 3, 3.50000000000000, 1.95312500000000)
4.5 (3.60156250000000, 3.43750000000000, 3.93750000000000, 4.50000000000000, 3.61328125000000)
5.5 (4.91601562500000, 3.59375000000000, 4.09375000000000, 4.93750000000000, 4.12207031250000)
6.5 (5.52441406250000, 5.60156250000000, 5.18750000000000, 5.09375000000000, 4.86645507812500)


The times to use the mulligan are when being assigned the 1 mile loop with
35 or 45 minutes remaining, the 3 mile loop with less than 30 minutes
remaining, or with 45 or 55 minutes remaining, the 3.5 mile loop with less
than 35 minutes remaining or with 55 minutes remaining, and the 4.5 mile
loop with less than 45 minutes remaining.

Simulations
-----------
[Numerical simulations](20250822.go) agree:

    $ go run 20250822.go
    4.866670
    5.351769